Adding kubectl.kubernetes.io/restartedAt annotation to kubernetes documentation (#40431)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

Author: sure @sftim, thanks for noticing, i will fix it.

One of the [recommended labels](/docs/concepts/overview/working-with-objects/common-labels/#labels).

### kubectl.kubernetes.io/restartedAt

> Example: `kubectl.kubernetes.io/restartedAt: "2023-03-31T15:30:00Z"`

Reviewer: ```suggestion
Example: `kubectl.kubernetes.io/restartedAt: '2023-03-31T15:30:00Z'`
```
nit: changed to single quotes as that's what I saw on my kubeadm v1.26 cluster

> Used on: Pod

Reviewer 2: Per my understanding,  this annotation could be applied to DaemonSets, Deployments, StatefulSets too by kubectl when you execute the `kubectl rollout restart` command. However, I'll let a tech reviewer from SIG CLI confirm this.

Reviewer: Tested this on a deployment, the Pod had the following annotation added:
`kubectl.kubernetes.io/restartedAt: '2023-04-04T14:07:20Z'`

> In this example, the annotation indicates that the Pod was most recently restarted at 3:30 PM UTC on March 31st, 2023.

Reviewer 2: I don't think this is required. However, I'm not against its addition.

Author: so what's your thoughts on this, i will remove these lines or leave as it is!

Reviewer 3: ```suggestion
In this example, the annotation indicates that the Pod is part of a workload where the last triggered restart
occurred at 15:30 (UTC) on March 31st, 2023.
```

> This annotation is used by kubernetes to keep track of the timestamp of the most recent container restart for a particular pod. This information can be useful for debugging and monitoring purposes.

Reviewer 2: Following the trend set by some of the other entries, this section could include which commands would trigger the addition of this annotation.

> This annotation is used by kubernetes to keep track of the timestamp of the most recent container restart for a particular pod. This information can be useful for debugging and monitoring purposes.

Reviewer 3: ```suggestion
The `kubectl` command line tool sets this annotation on pod templates (for example, within a Deployment)
as a way to trigger a restart.
See [`kubectl rollout restart`](/docs/reference/generated/kubectl/kubectl-commands#-em-restart-em-).
```

> The `"kubectl.kubernetes.io/restartedAt"` annotation is automatically added to the spec template by kubectl when the workload is restarted. It records the time of the most recent workload restart as a Unix timestamp.

Reviewer 3: ```suggestion
```
The text I'm removing is incorrect.

> The `"kubectl.kubernetes.io/restartedAt"` annotation is automatically added to a Pod's metadata by the kubelet when the Pod is restarted. It records the time that the most recent restart of the Pod occurred, as a Unix timestamp.

Reviewer 3: I don't think the kubelet adds this annotation; I think `kubectl` adds it (under specific circumstances)

> The `"kubectl.kubernetes.io/restartedAt"` annotation is automatically added to a Pod's metadata by the kubelet when the Pod is restarted. It records the time that the most recent restart of the Pod occurred, as a Unix timestamp.

Author: @sftim, Based on my research The `kubectl.kubernetes.io/restartedAt` annotation is added by the `kubelet` component when a pod is restarted. You can use this annotation to track and monitor the restart history of your Pods, which can be helpful for troubleshooting and identifying potential issues.

Here's an example of how you can retrieve the `restartedAt` annotation value for a Pod in Kubernetes using the `kubectl` command:

`kubectl describe pod <pod_name> | grep "restartedAt"`

Annotations:  kubectl.kubernetes.io/restartedAt: 2023-03-31T16:40:36.044Z

Hope now you can approve this PR!

> The `"kubectl.kubernetes.io/restartedAt"` annotation is automatically added to a Pod's metadata by the kubelet when the Pod is restarted. It records the time that the most recent restart of the Pod occurred, as a Unix timestamp.

Reviewer 2: @Biswajit-Mohapatra2 : Per [this codebase entry](https://github.com/kubernetes/kubectl/blob/3874cf79897cfe1e070e592391792658c44b78d4/pkg/polymorphichelpers/objectrestarter.go#L32), it is kubectl that adds the annotation to the Pod template so that the underlying Pods get reconciled via new ReplicaSets without any downtime. Please make the necessary amendments.

> The `"kubectl.kubernetes.io/restartedAt"` annotation is automatically added to a Pod's metadata by the kubectl when the Pod is restarted. It records the time that the most recent restart of the Pod occurred, as a Unix timestamp.

Reviewer 2: As aforementioned, this can be applied to other workload objects. If the tech review confirms that this is indeed the case, we need to generalize the statement and not make it specific to a pod.

```suggestion
The `"kubectl.kubernetes.io/restartedAt"` annotation is automatically added to the spec template by kubectl when the workload is restarted. It records the time of the most recent workload restart as a Unix timestamp.
```
